src/sinks/websocket/sink.rs

```diff
@@ -196,20 +196,26 @@ pub struct WebSocketSink {
     connector: WebSocketConnector,
     ping_interval: Option<NonZeroU64>,
     ping_timeout: Option<NonZeroU64>,
+    encode_as_binary: bool,
 }
 
 impl WebSocketSink {
     pub fn new(config: &WebSocketSinkConfig, connector: WebSocketConnector) -> crate::Result<Self> {
         let transformer = config.encoding.transformer();
         let serializer = config.encoding.build()?;
         let encoder = Encoder::<()>::new(serializer);
+        let encode_as_binary = match config.encoding.config() {
+            codecs::encoding::SerializerConfig::RawMessage => true,
+            _ => false,
+        };
```

Comment:
I think you can get everything you need closer to where it is needed, so no need for this stuff.

Before the `loop` call in `handle_events` , you could add:

```Rust
    let encode_as_binary = match self.encoder.serializer() {
        codecs::encoding::Serializer::RawMessage(_) => true,
        _ => false,
    };
````

Reply:
updated